pkg/security/loaders/raiffeisench/raiffeisench.go

```diff
@@ -0,0 +1,156 @@
+		name:  name,
+		isin:  isin,
+		valor: valor,
+	}
+}
+
+func getValorFromISIN(isin string) (int, error) {
+	// https://en.wikipedia.org/wiki/Valoren_number
+	// https://www.isin.org/isin-format/
+
+	if !strings.HasPrefix(isin, "CH") {
+		return 0, fmt.Errorf("valor number can only be extracted for Swiss financial instruments")
+	}
+
+	// strip CH
+	val := strings.TrimPrefix(isin, "CH")
+
+	// strip all leading zeros
+	val = strings.TrimLeft(val, "0")
+
+	// drop last digit (ISIN checksum)
+	val = val[0 : len(val)-1]
+
+	valor, err := strconv.Atoi(val)
+	if err != nil {
+		return 0, fmt.Errorf("failed to read valor number from ISIN: %w", err)
+	}
+	return valor, nil
+}
+
+func (r *RaiffeisenchQuoteLoader) Name() string {
+	return r.name
+}
+
+func (r *RaiffeisenchQuoteLoader) ISIN() string {
+	return r.isin
+}
+
+func (r *RaiffeisenchQuoteLoader) Valor() int {
+	return r.valor
+}
+
+type HistoryQuotesRequest struct {
+	Valor      int       `json:"valor"`
+	ExchangeId int       `json:"exchangeId"`
+	CurrencyId int       `json:"currencyId"`
+	From       time.Time `json:"from"`
+	To         time.Time `json:"to"`
+}
+
+type HistoryQuotesResponse struct {
+	HistoryQuotes []struct {
+		Date  string      `json:"date"`
+		Close float64     `json:"close"`
+		Open  interface{} `json:"open"`
+		High  interface{} `json:"high"`
+		Low   interface{} `json:"low"`
+	} `json:"historyQuotes"`
+}
+
+func (r *RaiffeisenchQuoteLoader) LoadQuotes() ([]security.Quote, error) {
+	// Build request payload
+	endDate := time.Now()
+	startDate := time.Now().Add(-1 * time.Hour * 24 * 365) // one year ago
+	payload := HistoryQuotesRequest{
+		Valor:      r.valor,
+		ExchangeId: 3233,
+		CurrencyId: 1, // CHF
+		From:       startDate.UTC(),
+		To:         endDate.UTC(),
+	}
+
+	payloadBytes, err := json.Marshal(payload)
+	if err != nil {
+		return nil, fmt.Errorf("error marshaling request body: %w", err)
+	}
+
+	// Build request
+	req, _ := http.NewRequest(
+		"POST",
+		"https://boerse.raiffeisen.ch/api/HistoryQuotes",
+		bytes.NewBuffer(payloadBytes),
+	)
+	req.Header.Set("Accept", "application/json")
+	req.Header.Set("Content-type", "application/json")
+	// We need to directly set the key in the header map,
+	// because Go's stdlib canonicalizes all HTTP headers,
+	// but the remote API requires this header to be lowercase.
+	req.Header["customer"] = []string{"raiffeisen-prod"}
+
+	// fmt.Println(req)
+
+	// Create an HTTP client and send the request
+	client := &http.Client{}
+	res, err := client.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("error during get request: %w", err)
+	}
+
+	if res.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("server replied with unexpected status code '%s'", res.Status)
+	}
+	// fmt.Println(res.StatusCode)
+	// fmt.Println(res.Header)
+
+	// Read and parse response body
+	var result HistoryQuotesResponse
+	reader := res.Body
+	// reader = io.TeeReader(res.Body, os.Stdout)
+	err = json.NewDecoder(reader).Decode(&result)
```

Comment:
```suggestion
	err = json.NewDecoder(res.Body).Decode(&result)
```

Reply:
Same here, TeeReader can be really helpful for debugging.